api-gateway/src/main/java/io/eventuate/examples/tram/sagas/ordersandcustomers/apiagateway/customers/OrderHistoryHandlers.java

```diff
@@ -0,0 +1,46 @@
+package io.eventuate.examples.tram.sagas.ordersandcustomers.apiagateway.customers;
+
+import io.eventuate.examples.tram.sagas.ordersandcustomers.apiagateway.proxies.CustomerNotFoundException;
+import io.eventuate.examples.tram.sagas.ordersandcustomers.apiagateway.orders.OrderResponse;
+import io.eventuate.examples.tram.sagas.ordersandcustomers.apiagateway.proxies.CustomerServiceProxy;
+import io.eventuate.examples.tram.sagas.ordersandcustomers.apiagateway.proxies.OrderServiceProxy;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+import static org.springframework.web.reactive.function.BodyInserters.fromObject;
+
+public class OrderHistoryHandlers {
+
+  private OrderServiceProxy orderService;
+  private CustomerServiceProxy customerService;
+
+  public OrderHistoryHandlers(OrderServiceProxy orderService, CustomerServiceProxy customerService) {
+    this.orderService = orderService;
+    this.customerService = customerService;
+  }
+
+  public Mono<ServerResponse> getOrderHistory(ServerRequest serverRequest) {
+    String customerId = serverRequest.pathVariable("customerId");
+
+    Mono<CustomerResponse> customer = customerService.findCustomerById(customerId);
+
+    Mono<List<OrderResponse>> orders = orderService.findOrdersByCustomerId(customerId);
+
+    return Mono
+            .zip(customer, orders)
+            .map(objects -> {
+              CustomerResponse c = objects.getT1();
+              c.setOrders(objects.getT2());
```

Comment:
This is ugly. `CustomerResponse` is what comes from the `CustomerService` 
This should create an `GetOrderHistoryResponse` - currently in ends to end tests - it should be a api-gateway-web-api module.

Reply:
Ok, I will fix